dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -1594,6 +1595,90 @@ abstract class DLCWallet
     } yield refundTx
   }
 
+  override def createMutualClose(
+      contractId: ByteVector,
+      localPayout: CurrencyUnit,
+      remotePayout: CurrencyUnit,
+      closeLocktime: UInt32): Future[DLCMutualCloseTLV] = {
+    dlcDAO.findByContractId(contractId).flatMap {
+      case None =>
+        Future.failed(
+          new RuntimeException(
+            s"Could not find a DLC with contract id ${contractId.toHex}"))
+      case Some(dlcDb) =>
+        for {
+          fundingInputs <- dlcInputsDAO.findByDLCId(dlcDb.dlcId)
+          scriptSigParams <- getScriptSigParams(dlcDb, fundingInputs)
+          signer <- dlcDataManagement
+            .signerFromDb(dlcId = dlcDb.dlcId,
+                          transactionDAO = transactionDAO,
+                          fundingUtxoScriptSigParams = scriptSigParams,
+                          keyManager = keyManager)
+            .map(_.get)
+
+          (offerPayout, acceptPayout) =
+            if (dlcDb.isInitiator) (localPayout, remotePayout)
+            else (remotePayout, localPayout)
+
+          tlv = signer.createMutualClose(offerPayout,
+                                         acceptPayout,
+                                         Vector.empty,
+                                         closeLocktime)
+
+          _ <- updateDLCState(contractId, DLCState.MutuallyClosed)
+        } yield tlv
+    }
+  }
+
+  override def closeDLC(closeTLV: DLCMutualCloseTLV): Future[Transaction] = {
+    val contractId = closeTLV.contractId
+    dlcDAO.findByContractId(contractId).flatMap {
+      case None =>
+        Future.failed(
+          new RuntimeException(
+            s"Could not find a DLC for contract id ${contractId.toHex}"))
+      case Some(dlcDb) =>
+        val dlcId = dlcDb.dlcId
+
+        for {
+          fundingInputs <- dlcInputsDAO.findByDLCId(dlcId)
+          scriptSigParams <- getScriptSigParams(dlcDb, fundingInputs)
+          signer <- dlcDataManagement
+            .signerFromDb(dlcId = dlcId,
+                          transactionDAO = transactionDAO,
+                          fundingUtxoScriptSigParams = scriptSigParams,
+                          keyManager = keyManager)
+            .map(_.get)
+
+          closeTx = signer.completeMutualClose(closeTLV)
+          _ = logger.info(
+            s"Created DLC refund transaction ${closeTx.txIdBE.hex} for contract ${contractId.toHex}")
```

Comment:
I'm guessing this should be `close` instead of `refund`

Reply:
thanks fixed